tests/test-support/src/lib.rs

```diff
@@ -316,13 +328,19 @@ impl RuntimeProvider for MockProvider {
             .lock()
             .unwrap()
             .push((server_addr.ip(), Protocol::Udp));
-        Box::pin(ready(Ok(MockUdpSocket::new(self.handler.clone()))))
+        Box::pin(ready(Ok(MockUdpSocket::new(
+            self.handler.clone(),
+            self.queries.clone(),
+        ))))
     }
 }
 
+type MockQueryCache = Arc<Mutex<HashMap<IpAddr, Vec<Query>>>>;
```

Comment:
nit: for top-down ordering define this below `MockUdpSocket` + `MockTcpStream` since both ref it.